pkg/engine/planner/physical/planner.go

```diff
@@ -270,17 +270,16 @@ func (p *Planner) processSelect(lp *logical.Select, ctx *Context) ([]Node, error
 	return []Node{node}, nil
 }
 
-// Convert [logical.Sort] into one [SortMerge] node.
+// Convert [logical.Sort] into one [NoOp] node.
+// It is temporarily used to pass the direction of the logical plan node to its children
+// and can be removed in the optimisation phase.
 func (p *Planner) processSort(lp *logical.Sort, ctx *Context) ([]Node, error) {
 	order := DESC
 	if lp.Ascending {
 		order = ASC
 	}
-	node := &SortMerge{
-		Column: &ColumnExpr{Ref: lp.Column.Ref},
-		Order:  order,
-	}
 
+	node := &NoOp{}
```

Comment:
Hm, I'm not sure this is necessary? Can't we still pass down the order to the children but then `return p.process(lp.Table, ctx.WithDirection(order)`?

Reply:
I'm not sure I'm following why that's an issue yet, but I might still be waking up. 

If we were to just propagate `return p.process(lp.Table, ctx.WithDirection(order)` up, I don't think we'd need to create an edge specifically for processSort? 

* If Sort was the final instruction, then the processed inner table (Merge) becomes the root node.   
* If Sort was not the final instruction, then the parent instruction (Limit) creates the edge directly to the inner table (Merge) 

Am I missing something there?

Reply:
Okay, that was a knot in my brain :man_facepalming: